core/src/test/java/dev/streamx/cli/license/LicenseAcceptingTest.java

```diff
@@ -0,0 +1,211 @@
+import static dev.streamx.cli.license.LicenseWiremockConfigs.StandardWiremockLicense.LICENSE_URL;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import dev.streamx.cli.license.LicenseTestProfiles.AcceptingLicenseTestProfile;
+import dev.streamx.cli.license.LicenseTestProfiles.ProceedingTestProfile;
+import dev.streamx.cli.license.model.LastLicenseFetch;
+import dev.streamx.cli.license.model.LicenseApproval;
+import dev.streamx.cli.license.model.LicenseSettings;
+import dev.streamx.cli.settings.SettingsStore;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import jakarta.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+@QuarkusTest
+@TestProfile(AcceptingLicenseTestProfile.class)
+class LicenseAcceptingTest {
+
+  public static final String OLD_URL = "http://old.streamx.dev/license.html";
+  public static final String OLD_NAME = "oldLicense";
+
+  LicenseArguments licenseArguments = new LicenseArguments();
+
+  @Inject
+  SettingsStore settingsStore;
+
+  @Inject
+  LicenseProcessorEntrypoint entrypoint;
+
+  @AfterEach
+  void shutdown() {
+    clearAcceptLicenceFlag();
+    clearSettings();
+  }
+
+  @Test
+  void shouldAcceptLicenseForClearEnvironment() {
+    // given
+    respondWithAcceptationIfUserIsAskedForAcceptance();
+
+    // when
+    entrypoint.process();
+
+    // then
+    LicenseSettings settings = verifyExistsLicense();
+    verifyLastFetchedLicense(settings, LICENSE_NAME, LICENSE_URL);
+    verifySingleApprovedLicense(settings, LICENSE_NAME, LICENSE_URL);
+  }
+
+  @Test
+  void shouldAutomaticallyAcceptLicenseIfFlagAcceptLicenseWasGiven() {
+    // given
+    respondWithAcceptationIfUserIsAskedForAcceptance();
+    licenseArguments.propagateAcceptLicense(true);
+
+    // when
+    entrypoint.process();
+
+    // then
+    LicenseSettings settings = verifyExistsLicense();
+    verifyLastFetchedLicense(settings, LICENSE_NAME, LICENSE_URL);
+    verifySingleApprovedLicense(settings, LICENSE_NAME, LICENSE_URL);
+  }
+
+  @Test
+  void shouldAcceptLicenseForPreviouslyRejectedLicense() {
+    // given
+    respondWithAcceptationIfUserIsAskedForAcceptance();
+
+    LocalDateTime now = LocalDateTime.now();
+    prepareEnvironmentWithNoAcceptedLicense(now);
+
+    // when
+    entrypoint.process();
+
+    // then
+    LicenseSettings settings = verifyExistsLicense();
+    verifyLastFetchedLicense(settings, LICENSE_NAME, LICENSE_URL);
+    verifySingleApprovedLicense(settings, LICENSE_NAME, LICENSE_URL);
+  }
+
+  @Test
+  void shouldSkipAcceptanceProceedingIfLastLicenseDataWasRecentlyRefreshed() {
```

Comment:
Is this test really checking if acceptance is skipped or if license data refresh is skipped?